RealTimeFaceMaskDetector/DateTimeDLL/DateTime.cpp

```diff
@@ -0,0 +1,162 @@
+#include "pch.h"
+#include <sstream>
+#include <iomanip>
+#include "DateTime.h"
+
+#define FORMAT_TWODIGIT_OUTPUT std::setw(2) << std::setfill('0')
+
+std::string DateTime::to_string() const {
+	std::stringstream result;
+	result << year << "-";
+	result << FORMAT_TWODIGIT_OUTPUT << month << "-";
+	result << FORMAT_TWODIGIT_OUTPUT << day << " ";
+	result << FORMAT_TWODIGIT_OUTPUT << hour << ":";
+	result << FORMAT_TWODIGIT_OUTPUT << minutes << ":";
+	result << FORMAT_TWODIGIT_OUTPUT << seconds;
+	return result.str();
+}
+
+void DateTime::parse() {
+	check_pattern();
+	year = std::stoi(datetime.substr(0, 4));
+	parse_month();
+	parse_day();
+	parse_hour();
+	parse_minutes();
+	parse_seconds();
+}
+
+void DateTime::check_pattern() {
+	if (!std::regex_search(datetime, datetime_pat)) {
+		throw DateTimeException("A format of datetime is invalid");
+	}
+}
+
+void DateTime::parse_month() {
+	month = std::stoi(datetime.substr(5, 7));
+
+	if (month > 12) {
+		throw DateTimeException("Month is invalid");
+	}
+}
+
+void DateTime::parse_day() {
+	day = std::stoi(datetime.substr(8, 10));
+
+	if (day > 31) {
+		throw DateTimeException("Day is invalid");
+	}
+}
+
+void DateTime::parse_hour() {
+	hour = std::stoi(datetime.substr(11, 13));
+
+	if (hour > 24) {
+		throw DateTimeException("Hour is invalid");
+	}
+}
+
+void DateTime::parse_minutes() {
+	minutes = std::stoi(datetime.substr(14, 16));
+
+	if (minutes > 60) {
+		throw DateTimeException("Value of minutes is invalid");
+	}
+}
+
+void DateTime::parse_seconds() {
+	seconds = std::stoi(datetime.substr(17, 19));
+
+	if (seconds > 60) {
+		throw DateTimeException("Value of seconds is invalid");
+	}
+}
+
+bool DateTime::operator<(const DateTime& datetime) {
+	return year < datetime.year || year == datetime.year && \
```

Comment:
Looks so weird, maybe split into inline functions will be better. In the other operators too

Reply:
I'm not sure that this will improve readability